sdk/go/pulumi-language-go/language_test.go

```
@@ -0,0 +1,269 @@
+	"l2-resource-alpha":                  "missing go.mod",
+	"l2-resource-simple":                 "missing go.mod",
+	"l2-resource-asset-archive":          "missing go.mod",
+	"l2-failed-create-continue-on-error": "missing go.mod",
+	"l2-engine-update-options":           "missing go.mod",
+}
+
+func TestLanguage(t *testing.T) {
+	t.Parallel()
+
+	engineAddress, engine := runTestingHost(t)
+
+	tests, err := engine.GetLanguageTests(context.Background(), &testingrpc.GetLanguageTestsRequest{})
+	require.NoError(t, err)
+
+	cancel := make(chan bool)
+
+	// Run the language plugin
+	handle, err := rpcutil.ServeWithOptions(rpcutil.ServeOptions{
+		Init: func(srv *grpc.Server) error {
+			host := newLanguageHost(engineAddress, "", "")
+			pulumirpc.RegisterLanguageRuntimeServer(srv, host)
+			return nil
+		},
+		Cancel: cancel,
+	})
+	require.NoError(t, err)
+
+	// Create a temp project dir for the test to run in
+	rootDir := t.TempDir()
+
+	snapshotDir := "./testdata/"
+
+	// Prepare to run the tests
+	prepare, err := engine.PrepareLanguageTests(context.Background(), &testingrpc.PrepareLanguageTestsRequest{
+		LanguagePluginName:   "go",
+		LanguagePluginTarget: fmt.Sprintf("127.0.0.1:%d", handle.Port),
+		TemporaryDirectory:   rootDir,
+		SnapshotDirectory:    snapshotDir,
+		CoreSdkDirectory:     "../..",
+		CoreSdkVersion:       sdk.Version.String(),
+		SnapshotEdits: []*testingrpc.PrepareLanguageTestsRequest_Replacement{ /*
+				{
+					Path:        "go\\.mod",
+					Pattern:     fmt.Sprintf("pulumi-pulumi-%s\\.tgz", sdk.Version.String()),
+					Replacement: "pulumi-pulumi-CORE.VERSION.tgz",
+				},*/
+			{
+				Path:        "go\\.mod",
+				Pattern:     rootDir + "/artifacts",
+				Replacement: "/ROOT/artifacts",
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	for _, tt := range tests.Tests {
+		tt := tt
+		t.Run(tt, func(t *testing.T) {
+			t.Parallel()
+
+			if expected, ok := expectedFailures[tt]; ok {
+				t.Skipf("Skipping known failure: %s", expected)
+			}
```

Should we run the tests and have a `require.ErrorContains` with the expected failure output instead of skipping them?

---

No I think better to just skip them for now. We should get them working fairly quickly.